app/frontend/app/controllers/group.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
+import HasCurrentUser from '../mixins/has-current-user';
+/* global Messenger */
+
+export default Ember.Controller.extend(HasCurrentUser, {
+  // this flag is used so that the "Leave Group" button doesn't switch
+  // to "Join Group" when waiting for the server to respond.
+  loading: false,
+
+  coverUpload: Ember.Object.create(),
```

Comment:
Let's not duplicate the cover upload code, extract it from here and users into a mixin.